src/NHibernate/Transaction/AdoNetWithDistributedTransactionFactory.cs

```diff
@@ -205,6 +208,8 @@ void IEnlistmentNotification.InDoubt(Enlistment enlistment)
 				{
 					sessionImplementor.ConnectionManager.AfterTransaction();
```

Comment:
In Hibernate these TWO calls are combined in `TransactionCoordinator.AfterTransactionCompletion`